iroh-gateway/src/core.rs

```diff
@@ -37,10 +36,11 @@ pub struct Core {
 }
 
 #[derive(Debug)]
-pub(crate) struct State {
+pub struct State {
```

Comment:
why does this need to be public?

Reply:
Kinda has to do with the first comment on the PR. Build complains that I'm leaking crate private `State` into `Client` by passing it in which is it's parent. It's purely a workaround. Ideally I should split those off and have core be the owner of both `State` and `Client` independently.